### PR TITLE
slider мерч

### PR DESCRIPTION
**Pull request by mar1cruz**

### CODE REVIEW

**Review by makcik303123 — APPROVED** [review]

File: commercialNetwork/src/js/initSlider.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    return `translate3d(${index * shift}px, 0px, 0px)`
+  }
+
+  function shiftCalculate() {
+    const itemWidth = items[0].offsetWidth
+    return (shiftValue = parseInt(getGap.gap) + itemWidth)
+  }
+
+  function getRemoveSlides(selector, window, slides) {
+    if (selector === "#slider-news" && window >= 768 && window <= 992) {
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js:174]: не должно быть привязки к slider-news

File: commercialNetwork/src/js/initSlider.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  const btnNext = slider.querySelector(".slider-button__right")
+  const btnPrev = slider.querySelector(".slider-button__left")
+  const track = slider.querySelector(".slider-track")
+  const items = slider.querySelectorAll(".slider-item")
+  const getGap = getComputedStyle(track)
+  const slides = param.removeSlides
+  let text = slider.querySelectorAll(".text")
+  if (!text.length) text = false
+
+  console.log(text)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js:12]: remove logs

File: commercialNetwork/src/js/initSlider.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    if (indexActiveSlide !== 0) btnPrev.classList.remove("hiddenBtn-left")
+
+    setTimeout(() => {
+      document.body.classList.remove("animation")
+    }, 300)
+  })
+
+  btnPrev.addEventListener("click", () => {
+    const zdvig = shiftCalculate()
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove zdvig

File: commercialNetwork/src/js/initSlider.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    const countActiveSlide = activeSlides(slider)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: rename to getTotalVisibleSlides

File: commercialNetwork/src/js/initSlider.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+  function shift(index, shift) {
+    return `translate3d(${index * shift}px, 0px, 0px)`
+  }
+
+  function shiftCalculate() {
+    const itemWidth = items[0].offsetWidth
+    return (shiftValue = parseInt(getGapTrack.gap) + itemWidth)
+  }
+
+  function redraw(direction, text, indexActiveSlide) {
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove redraw function

File: commercialNetwork/src/js/initSlider.js [hunk 6]
```diff
@@ -0,0 +1,201 @@
+function initSlider(selector) {
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: add default params with gap: 20

File: commercialNetwork/src/js/initSlider.js [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+  const slider = document.querySelector(selector)
+  const btnNext = slider.querySelector(".slider-button__right")
+  const btnPrev = slider.querySelector(".slider-button__left")
+  const track = slider.querySelector(".slider-track")
+  const items = slider.querySelectorAll(".slider-item")
+  const getGapTrack = getComputedStyle(track)
+
+  let text = slider.querySelectorAll(".text")
+  if (!text.length) text = false
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove

File: commercialNetwork/src/js/initSlider.js [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+  let indexActiveSlide = 0
+  let positionStart = 0
+  let currentPosition = 0
+  let isDrag = false
+
+  if (indexActiveSlide === 0) btnPrev.classList.add("hiddenBtn-left")
+
+  btnNext.addEventListener("click", () => {
+    const zdvig = shiftCalculate()
+    const countActiveSlide = activeSlides(slider)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove

File: commercialNetwork/src/js/initSlider.js [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+
+    if (Math.abs(indexActiveSlide) <= items.length - countActiveSlide) {
+      indexActiveSlide--
+      track.style.transform = shift(indexActiveSlide, zdvig)
+      document.body.classList.add("animation")
+
+      if (text) redraw("next", text, indexActiveSlide)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: change to call function changeSlide

File: commercialNetwork/src/js/initSlider.js [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+      positionStart > positionEnd &&
+      Math.abs(positionEnd - positionStart) > zdvig * 0.2 &&
+      -indexActiveSlide !== items.length - countActiveSlide
+    ) {
+      if (
+        positionStart - positionEnd - 2 * zdvig > 0 &&
+        Math.abs(indexActiveSlide) + 2 < items.length - countActiveSlide
+      ) {
+        indexActiveSlide -= 2
+        if (text) redraw("next", text, indexActiveSlide)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: change to changeSlide()

File: commercialNetwork/src/js/initSlider.js [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+      } else {
+        indexActiveSlide--
+        if (text) redraw("next", text, indexActiveSlide)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: change to changeSlide()

File: commercialNetwork/src/js/initSlider.js [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+      }
+
+      document.body.classList.add("animation")
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove

File: commercialNetwork/src/js/initSlider.js [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    if (positionStart < positionEnd && positionEnd - positionStart > zdvig * 0.2 && indexActiveSlide !== 0) {
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: const ... = positionEnd - positionStart > zdvig * 0.2
const ... = indexActiveSlide !==
 0.2 is constant

File: commercialNetwork/src/js/initSlider.js [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+      if (Math.abs(positionStart - positionEnd) - 2 * zdvig > 0 && indexActiveSlide + 2 <= 0) {
+        indexActiveSlide += 2
+        if (text) redraw("prev", text, indexActiveSlide)
+      } else {
+        indexActiveSlide++
+        if (text) redraw("prev", text, indexActiveSlide)
+      }
+
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: change to changeSlide()

File: commercialNetwork/src/js/initSlider.js [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+      document.body.classList.add("animation")
+    }
+
+    setTimeout(() => {
+      document.body.classList.remove("animation")
+    }, 300)
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove

File: commercialNetwork/src/js/initSlider.js [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+    if (positionStart < positionEnd && positionEnd - positionStart > zdvig * 0.2 && indexActiveSlide !== 0) {
+      if (Math.abs(positionStart - positionEnd) - 2 * zdvig > 0 && indexActiveSlide + 2 <= 0) {
+        indexActiveSlide += 2
+        if (text) redraw("prev", text, indexActiveSlide)
+      } else {
+        indexActiveSlide++
+        if (text) redraw("prev", text, indexActiveSlide)
+      }
+
+      document.body.classList.add("animation")
```
Comment (stillgg) [inline commercialNetwork/src/js/initSlider.js]: remove

**Review by stillgg — COMMENTED** [review]
- remove comments
- rename initSlider to slider
- Max's slider moves to slider.js

File: commercialNetwork/src/js/initSlider.js (hunk 16 above)
Comment (stillgg) [reply commercialNetwork/src/js/initSlider.js]: addEventListener('resize') on this event you need to call initSlide again

**Review by stillgg — COMMENTED** [review]
mobile slider team doesn't work

File: commercialNetwork/src/js/slider.js [hunk 17]
```diff
@@ -1,47 +1,161 @@
...
+      btnNext.classList.add("hidden")
+    }
+
+    currentPosition = itemWidth * -indexActiveSlide
+
+    if (indexActiveSlide !== 0) btnPrev.classList.remove("hidden")
+  })
+
+  btnPrev.addEventListener("click", () => {
+    const countActiveSlide = getTotalVisibleSlides(area)
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:35]: put countActiveSlide in slider scope

File: commercialNetwork/src/js/slider.js [hunk 18]
```diff
@@ -1,47 +1,161 @@
...
+
+    positionStart = event.type === "touchstart" ? event.touches[0].clientX : event.clientX
+    positionStart += currentPosition
   }
-  if (activeSlide === slides.length - 1) {
-    btnRight.classList.add("disabled");
+
+  function getTotalVisibleSlides(selector) {
+    const sliderWidth = selector.offsetWidth
+    const itemWidth = items[0].offsetWidth
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:63]: remove

File: commercialNetwork/src/js/slider.js [hunk 19]
```diff
@@ -1,47 +1,161 @@
...
+    return Math.floor(sliderWidth / itemWidth)
+  }
+
+  function onDragOver(event) {
+    if (!isDrag) return
+
+    const move = event.type === "touchmove" ? event.touches[0].clientX : event.clientX
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:70]: shorter expression : `const move = event.touches ? touches[0].clientX:event.clientX`:

File: commercialNetwork/src/js/slider.js [hunk 20]
```diff
@@ -1,47 +1,161 @@
...
+
+    track.style.transform = `translate3d(${move - positionStart}px, 0px, 0px)`
+
+    track.style.transitionDuration = "0ms"
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:74]: put this code in onDragStart function

File: commercialNetwork/src/js/slider.js [hunk 21]
```diff
@@ -1,47 +1,161 @@
...
+
+    if (Math.abs(indexSlide) === items.length - countActiveSlide) {
+      btnNext.classList.add("hidden")
+    } else btnNext.classList.remove("hidden")
+  }
+
+  function onDragEnd(event) {
+    if (!isDrag) return
+
+    let positionEnd = event.type === "touchend" ? event.changedTouches[0].clientX : event.clientX
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:96]: - const
- shorter expression

File: commercialNetwork/src/js/slider.js [hunk 22]
```diff
@@ -1,47 +1,161 @@
...
+
+    positionEnd += currentPosition
+
+    const countActiveSlide = getTotalVisibleSlides(area)
+    const procentShift = 0.2
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:101]: bulshit name 
Screenshot 2023-08-07 at 18 43 28

File: commercialNetwork/src/js/slider.js [hunk 23]
```diff
@@ -1,47 +1,161 @@
...
+        indexActiveSlide -= 2
+      } else {
+        indexActiveSlide--
+      }
+    }
+
+    if (positionStart < positionEnd && isMoved && isFirstSlide) {
+      if (Math.abs(positionStart - positionEnd) - 2 * itemWidth > 0 && indexActiveSlide + 2 <= 0) {
+        indexActiveSlide += 2
+      } else {
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:123]: remove

File: commercialNetwork/src/js/slider.js [hunk 24]
```diff
@@ -1,47 +1,161 @@
...
+    if (
+      positionStart > positionEnd &&
+      Math.abs(positionEnd - positionStart) > itemWidth * procentShift &&
+      Math.abs(indexActiveSlide) !== items.length - countActiveSlide
+    ) {
+      if (
+        positionStart - positionEnd - 2 * itemWidth > 0 &&
+        Math.abs(indexActiveSlide) + 2 < items.length - countActiveSlide
+      ) {
+        indexActiveSlide -= 2
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:114]: remove

File: commercialNetwork/src/js/slider.js [hunk 25]
```diff
@@ -1,47 +1,161 @@
...
+    flippButtons(indexActiveSlide)
+    changeSlide(indexActiveSlide)
+
+    track.style.transitionDuration = "400ms"
+
+    currentPosition = -(indexActiveSlide * itemWidth)
+    isDrag = false
+  }
+
+  function goingBeyond() {
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:137]: trash name

File: commercialNetwork/src/js/teamSlider.js [hunk 26]
```diff
@@ -0,0 +1,47 @@
...
+  // console.log(activeSlide);
+  btnDisabled()
+}
+const clickLeft = () => {
+  slides[activeSlide].classList.remove("slider__img__active")
+  slides[activeSlide].classList.add("slider__img__next")
+  slides[activeSlide - 1].classList.add("slider__img__active")
+  slides[activeSlide - 1].classList.remove("slider__img__prev")
+  activeSlide += -1
+  // console.log(activeSlide);
```
Comment (stillgg) [inline commercialNetwork/src/js/teamSlider.js:23]: wtf

File: commercialNetwork/src/js/teamSlider.js [hunk 27]
```diff
@@ -0,0 +1,47 @@
...
+
+let activeSlide = 0
+
+const clickRight = () => {
+  slides[activeSlide].classList.remove("slider__img__active")
+  slides[activeSlide].classList.add("slider__img__prev")
+  slides[activeSlide + 1].classList.add("slider__img__active")
+  slides[activeSlide + 1].classList.remove("slider__img__next")
+  activeSlide++
+  // console.log(activeSlide);
```
Comment (stillgg) [inline commercialNetwork/src/js/teamSlider.js:14]: wtf

File: commercialNetwork/src/js/slider.js [hunk 28]
```diff
@@ -1,47 +1,170 @@
...
+    sliderIndex: 0,
+    gap: 20,
+  },
+) {
+  const area = document.querySelector(selector)
+  const btnNext = area.querySelector(".slider-button__right")
+  const btnPrev = area.querySelector(".slider-button__left")
+  const track = area.querySelector(".slider-track")
+  const items = area.querySelectorAll(".slider-item")
+  const getGapTrack = params.gap
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:13]: getGapTrack it's a bad name for number variable, "get" means function

File: commercialNetwork/src/js/slider.js (hunk 21 above)
Comment (stillgg) [reply commercialNetwork/src/js/slider.js:87]: replace positionEnd

File: commercialNetwork/src/js/slider.js [hunk 29]
```diff
@@ -1,47 +1,170 @@
...
+    flippButtons(indexActiveSlide)
+    changeSlide(indexActiveSlide)
+
+    track.style.transitionDuration = "400ms"
+
+    currentPosition = -(indexActiveSlide * itemWidth)
+    isDrag = false
+  }
+
+  function checkExit() {
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:116]: onMouseLeave rename

File: commercialNetwork/src/js/slider.js [hunk 30]
```diff
@@ -1,47 +1,170 @@
...
+    document.removeEventListener("mouseleave", checkExit)
+
+    track.removeEventListener("mousedown", onDragStart)
+    document.removeEventListener("mousemove", onDragOver)
+    document.removeEventListener("mouseup", onDragEnd)
+
+    track.removeEventListener("touchstart", onDragStart)
+    document.removeEventListener("touchmove", onDragOver)
+    document.removeEventListener("touchend", onDragEnd)
+  }
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:133]: create method destroy
` const slider1 = slider(); slider1.destroy()`

File: commercialNetwork/src/js/slider.js [hunk 31]
```diff
@@ -1,47 +1,170 @@
...
+
+  function callSlider() {
+    params = {
+      gap: 20,
+      sliderIndex: indexActiveSlide,
+    }
+    slider(selector, params)
+  }
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:141]: remove

File: commercialNetwork/src/js/slider.js [hunk 32]
```diff
@@ -1,47 +1,170 @@
...
+
+  function updateSlider() {
+    clearSlider()
+    callSlider()
+  }
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:146]: remove

File: commercialNetwork/src/js/slider.js [hunk 33]
```diff
@@ -1,47 +1,170 @@
...
+
+  function debounceResize() {
+    clearTimeout(resizeTimeout)
+    resizeTimeout = setTimeout(updateSlider, 200)
+  }
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:151]: remove

File: commercialNetwork/src/js/slider.js [hunk 34]
```diff
@@ -1,47 +1,170 @@
...
+  const getGapTrack = params.gap
+  const itemWidth = items[0].clientWidth + getGapTrack
+  const countActiveSlide = getTotalVisibleSlides(area)
+
+  let indexActiveSlide = params.sliderIndex
+
+  let positionStart = 0
+  let currentPosition = 0
+  let isDrag = false
+  let resizeTimeout
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:22]: remove

File: commercialNetwork/src/js/slider.js [hunk 35]
```diff
@@ -1,47 +1,176 @@
...
+  params = {
+    sliderIndex: 0,
+    gap: 20,
+    effect: "none",
+  },
+) {
+  const defaultParams = {
+    sliderIndex: params?.sliderIndex || 0,
+    gap: params?.gap || 20,
+    effect: params.effect || "none",
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:12]: default

File: commercialNetwork/src/js/slider.js [hunk 36]
```diff
@@ -1,47 +1,176 @@
...
+
+  function getTotalVisibleSlides(selector) {
+    const sliderWidth = selector.clientWidth
+    const itemWidth = items[0].clientWidth
+    return Math.floor(sliderWidth / itemWidth)
+  }
+
+  function changeSlide(indexSlide) {
+    if (defaultParams.effect === "none") {
+      track.style.transform = `translate3d(-${indexSlide * itemWidth}px, 0px, 0px)`
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:56]: create function defaultEffect

File: commercialNetwork/src/js/slider.js [hunk 37]
```diff
@@ -1,47 +1,176 @@
...
+
+    if (indexSlide >= items.length - totalVisibleSlides) {
+      btnNext.classList.add("hidden")
+    } else btnNext.classList.remove("hidden")
+
+    let i = items.length
+    items.forEach((element) => {
+      element.style.zIndex = i
+      i--
+    })
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:71]: move to cardEffect

File: commercialNetwork/src/js/slider.js [hunk 38]
```diff
@@ -1,47 +1,176 @@
...
+      track.style.transform = `translate3d(-${indexSlide * itemWidth}px, 0px, 0px)`
+    }
+
+    if (indexSlide === 0) {
+      btnPrev.classList.add("hidden")
+    } else btnPrev.classList.remove("hidden")
+
+    if (indexSlide >= items.length - totalVisibleSlides) {
+      btnNext.classList.add("hidden")
+    } else btnNext.classList.remove("hidden")
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:65]: move to toggleButtons function

File: commercialNetwork/src/js/slider.js [hunk 39]
```diff
@@ -1,47 +1,176 @@
...
+    })
+
+    items[indexSlide - 1] && items[indexSlide - 1].classList.remove("prev")
+    items[indexSlide] && items[indexSlide].classList.remove("next") + items[indexSlide].classList.remove("prev")
+
+    items[prev].classList.remove("active")
+    items[indexSlide].classList.add("active")
+
+    items[indexSlide - 1] && items[indexSlide - 1].classList.add("prev")
+    items[indexSlide + 1] && items[indexSlide + 1].classList.add("next")
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:80]: replacre && to if

File: commercialNetwork/src/js/slider.js [hunk 40]
```diff
@@ -1,47 +1,176 @@
...
+
+  const track = area.querySelector(".slider-track")
+  const itemWidth = items[0].clientWidth + defaultParams.gap
+  const totalVisibleSlides = getTotalVisibleSlides(area)
+  const percentShift = 0.2
+
+  let indexActiveSlide = defaultParams.sliderIndex
+  let positionStart = 0
+  let isDrag = false
+  let prev = indexActiveSlide
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:28]: remove prev variable

File: commercialNetwork/src/js/slider.js [hunk 41]
```diff
@@ -1,47 +1,176 @@
...
+
+  function onDragEnd(event) {
+    if (!isDrag) return
+
+    const positionEnd = event.touches ? event.changedTouches[0].clientX : event.clientX
+    const isMoved = Math.abs(positionEnd - positionStart) > itemWidth * percentShift
+    const isFirstSlide = indexActiveSlide !== 0
+    const isLastSlide = indexActiveSlide !== items.length - totalVisibleSlides
+
+    prev = indexActiveSlide
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:110]: remove prev variable

File: commercialNetwork/src/js/slider.js [hunk 42]
```diff
@@ -1,47 +1,176 @@
...
+    btnNext.addEventListener("click", onClickBtnRight)
+    btnPrev.addEventListener("click", onClickBtnLeft)
+
+    track.addEventListener("mousedown", onDragStart)
+    document.addEventListener("mousemove", onDragOver)
+    document.addEventListener("mouseup", onDragEnd)
+
+    track.addEventListener("touchstart", onDragStart)
+    document.addEventListener("touchmove", onDragOver)
+    document.addEventListener("touchend", onDragEnd)
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:145]: add events if effect is default

**Comment by stillgg** [comment]
quotes fix 
Screenshot 2023-08-17 at 22 22 40

**Comment by stillgg** [comment]
replace images and text how on https://promotion.ru/#team

**Comment by stillgg** [comment]
slider is cropped it needs to fix 
Screenshot 2023-08-17 at 22 34 35

**Comment by stillgg** [comment]
fix this issue here https://github.com/stillgg/camp/issues/95

File: commercialNetwork/src/js/slider.js [hunk 43]
```diff
@@ -1,47 +1,192 @@
...
+  const track = area.querySelector(".slider-track")
+  const itemWidth = items[0].clientWidth + defaultParams.gap
+  const totalVisibleSlides = getTotalVisibleSlides(area)
+  const shiftRatio = 0.2
+
+  let indexActiveSlide = defaultParams.sliderIndex
+  let positionStart = 0
+  let isDrag = false
+
+  if (indexActiveSlide === 0) btnPrev.classList.add("hidden")
```
Comment (stillgg) [inline commercialNetwork/src/js/slider.js:29]: remove

**Comment by stillgg** [comment]
It didn't fix https://github.com/stillgg/camp/pull/87#issuecomment-1682851159

**Comment by stillgg** [comment]
remove btn arrows on mobile (max-width: 576.98px)

**Comment by stillgg** [comment]
https://github.com/stillgg/camp/assets/58826012/816d976a-05d7-4323-b465-e958d584519a
remove transform button animation

**Review by stillgg — APPROVED** [review]

**Pull request merged**